server-common/src/main/java/org/apache/kafka/server/group/share/NoOpShareStatePersister.java

```diff
@@ -26,10 +26,23 @@
 import java.util.concurrent.CompletableFuture;
 import java.util.stream.Collectors;
 
+/**
+ * A no-op singleton implementation of {@link Persister} interface.
+ */
 public class NoOpShareStatePersister implements Persister {
 
   private static final Logger log = LoggerFactory.getLogger(NoOpShareStatePersister.class);
+  private static volatile Persister instance = null;
```

Comment:
Isn't double check lock would be better?

Reply:
Implemented double-checked locks